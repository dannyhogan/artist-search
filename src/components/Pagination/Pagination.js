import React from 'react';
import PropTypes from 'prop-types';

export default function Pagination({ pageUp, pageDown, count = 1, page }) {

  const totalPages = Math.ceil(count / 25);
  const clickableDown = page > 1 ? false : true;
  const clickableUp = page >= totalPages ? true : false;
  if(count === 0) page = 0;
  
  return (
    <>
      <button disabled={clickableDown} onClick={() => pageDown()}>←</button>
      <label>Page {page} of {totalPages}</label>
      <button disabled={clickableUp} onClick={() => pageUp()}>→</button>
    </>
  );
}

Pagination.propTypes = {
  pageUp: PropTypes.func.isRequired,
  pageDown: PropTypes.func.isRequired,
  count: PropTypes.number,
  page: PropTypes.number
};
