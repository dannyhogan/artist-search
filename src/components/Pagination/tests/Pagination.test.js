import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../Pagination';

describe('Pagination component', () => {
  it('renders Pagination component', () => {
    const wrapper = shallow(<Pagination pageUp={() => {}} pageDown={() => {}} count={3} page={5} />);
    expect(wrapper).toMatchSnapshot();
  });
});
