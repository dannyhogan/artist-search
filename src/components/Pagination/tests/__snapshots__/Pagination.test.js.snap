// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders Pagination component 1`] = `
<Fragment>
  <button
    disabled={false}
    onClick={[Function]}
  >
    ←
  </button>
  <label>
    Page 
    5
     of 
    1
  </label>
  <button
    disabled={true}
    onClick={[Function]}
  >
    →
  </button>
</Fragment>
`;
